### PR TITLE
App-gateway-improve-docs

### PR DESCRIPTION
**Pull request by diegolagospagopa**
### List of changes

- added draw.io arch design
- added more comments and description
- best readme layout

### Motivation and context

Improve the documentation for the app gateway module

### Type of changes

- [ ] Add new module
- [ ] Update existing module
- [ ] Remove existing module
- [x] Other

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app_gateway/variables.tf [hunk 1]
```diff
@@ -35,44 +35,46 @@ variable "public_ip_id" {
...
-    firewall_policy_id = string
+    protocol           = string # The Protocol which should be used. Possible values are Http and Https
+    host               = string # The Hostname which should be used for this HTTP Listener. Setting this value changes Listener Type to 'Multi site'.
+    port               = number # The port used for this Frontend Port.
+    ssl_profile_name   = string # The name of the associated SSL Profile which should be used for this HTTP Listener.
+    firewall_policy_id = string # The ID of the Web Application Firewall Policy which should be used for this HTTP Listener.
     certificate = object({
-      name = string
-      id   = string
+      name = string #(Required) The Name of the SSL certificate that is unique within this Application Gateway
```
Comment (uolter) [inline app_gateway/variables.tf:60]:
```suggestion
      name = string  #The Name of the SSL certificate that is unique within this Application Gateway
```

File: app_gateway/variables.tf [hunk 2]
```diff
@@ -35,44 +35,46 @@ variable "public_ip_id" {
...
+      id   = string #(Optional) Secret Id of (base-64 encoded unencrypted pfx) Secret or Certificate object stored in Azure KeyVault. You need to enable soft delete for keyvault to use this feature. Required if data is not set.
```
Comment (uolter) [inline app_gateway/variables.tf:61]:
```suggestion
      id   = string # Secret Id of (base-64 encoded unencrypted pfx) Secret or Certificate object stored in Azure KeyVault. You need to enable soft delete for keyvault to use this feature. Required if data is not set.
```

**Review by uolter — CHANGES_REQUESTED** [review]

File: app_gateway/variables.tf (hunk 1 above)
Comment (diegolagospagopa) [reply app_gateway/variables.tf:60]: Done

File: app_gateway/variables.tf (hunk 2 above)
Comment (diegolagospagopa) [reply app_gateway/variables.tf:61]: Done

**Review by uolter — APPROVED** [review]

**Pull request merged**